### PR TITLE
Document Environment variables: Fix #2036

### LINKED ISSUE #2036: [documentation] undocumented environment variables
## problem
Some environment variables to configure `tox` are undocumented as far as I searched on [readthedocs](https://tox.readthedocs.io/en/latest/index.html).  Some examples are:
* `TOX_DISCOVER`, which is in [changelog 3.14.5](https://tox.readthedocs.io/en/latest/changelog.html#v3-14-5-2020-02-16) but not elsewhere.  I searched in this repository for an environment-variable alternative to `--discover` command line option to find this environment variable.
* `TOX_PROVISION`

and [others can be found by searching](https://github.com/tox-dev/tox/search?q=os+environ+get).

## possible improvement
1. There is an [Injected environment variables section in tox configuration specification](https://github.com/tox-dev/tox/blob/master/docs/config.rst#injected-environment-variables).  I found it is most straightforward to put them into this section.  Since they don't seem to be environment variables "that you can use to test *that* your command is running within tox", the paragraph can also be rewritten.
2. Since `TOX_DISCOVER` corresponds to `--discover`, it is also possible to add it to [`help` argument of `parser.add_argument` calls`](https://github.com/tox-dev/tox/blob/master/src/tox/config/__init__.py).  I haven't check if this applies to all the undocumented environment variables.

### PR DESCRIPTION
**Pull request by majiang**
This is a PR for #2036.

## Contribution checklist:

(also see [CONTRIBUTING.rst](../tree/master/CONTRIBUTING.rst) for details)

- [x] wrote descriptive pull request text
- [ ] ~~added/updated test(s)~~
- [x] updated/extended the documentation
- [x] added relevant [issue keyword](https://help.github.com/articles/closing-issues-using-keywords/)
      in message body
- [x] added news fragment in [changelog folder](../tree/master/docs/changelog)
  * fragment name: `<issue number>.<type>.rst` for example (588.bugfix.rst)
  * `<type>` is must be one of `bugfix`, `feature`, `deprecation`,`breaking`, `doc`, `misc`
  * if PR has no issue: consider creating one first or change it to the PR number after creating the PR
  * "sign" fragment with "by :user:`<your username>`"
  * please use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files - by :user:`superuser`."
  * also see [examples](../tree/master/docs/changelog)
- [x] added yourself to `CONTRIBUTORS` (preserving alphabetical order)

### CODE REVIEW

File: docs/config.rst [hunk 1]
```diff
@@ -1086,8 +1086,22 @@ user can bypass the system's interpreter parser by defining the
 When the workaround is enabled, all tox-invoked text file executables will have
 their interpreter directive parsed by and explicitly executed by ``tox``.
 
+Environment variables
+---------------------
+tox will treat the following environment variables:
```
Comment (jugmac00) [inline docs/config.rst:1091]: I think you need either a double colon or an empty line, so the list gets properly rendered - have a look at https://github.com/tox-dev/tox/blob/e28a4983c59c1470f8a4d9ee482d28157bd83bbb/docs/config.rst#environment-variables - currently this does not look good.

You can also build the docs locally on your machine before pushing via `tox -e docs`

File: docs/config.rst [hunk 2]
```diff
@@ -1086,8 +1086,22 @@ user can bypass the system's interpreter parser by defining the
...
+- ``TOX_DISCOVER`` for python discovery first try the python executables under these paths
```
Comment (jugmac00) [inline docs/config.rst:1092]:
```suggestion
- ``TOX_DISCOVER`` for Python discovery first try the python executables under these paths
```

File: docs/config.rst [hunk 3]
```diff
@@ -1086,8 +1086,22 @@ user can bypass the system's interpreter parser by defining the
...
+- ``TOXENV`` see envlist.
+- ``TOX_LIMITED_SHEBANG`` see Advanced settings > Handle interpreter directives with long lengths.
+- ``TOX_PARALLEL_NO_SPINNER`` see Parallel mode.
+- ``_TOX_PARALLEL_ENV`` lets tox know that it is invoked in the parallel mode.
+- ``TOX_PROVISION`` is only intended to be used internally.
+- ``TOX_REPORTER_TIMESTAMP`` enables showing for each output line its delta
+since the tox startup when set to ``1``.
```
Comment (jugmac00) [inline docs/config.rst:1099]: I do not understand this sentence :-/

**Review by jugmac00 — CHANGES_REQUESTED** [review]
Thank you for the pull request.

I have a couple of inline annotations.

Also, could you please link to the corresponding sections in the documenation whenever you write "see ..."?

Thank you!

File: docs/config.rst (hunk 3 above)
Comment (majiang) [reply docs/config.rst:1100]: Do you mean the sentence for `TOX_REPORTER_TIMESTAMP`?

Comment (jugmac00) [reply docs/config.rst:1100]: Yes :-/

**Pull request merged**